### PR TITLE
map fragment added

### PR DESCRIPTION
**Pull request by mananwason**
@creativepsyco Please see and merge

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -21,14 +21,15 @@ android {
...
     compile 'com.android.support:cardview-v7:22.2.0'
-    compile 'com.android.support:recyclerview-v7:+'
+    compile 'com.android.support:recyclerview-v7:22.2.0'
     compile 'com.google.code.gson:gson:2.3.1'
     compile 'com.squareup.okhttp:okhttp:2.0.0'
     compile 'com.squareup.retrofit:retrofit:1.9.0'
     compile 'com.android.support:design:22.2.0'
     compile 'com.squareup.picasso:picasso:2.5.2'
-
+    compile 'com.google.android.gms:play-services:7.5.0'
```
Comment (creativepsyco) [inline app/build.gradle]: Don't need to use the full Play services Lib. Its damn huge bloats up the APK. Instead use the `com.google.android.gms:play-services-base:7.5.0` followed by `com.google.android.gms:play-services-maps:7.5.0` See [Setting up Play Services][1] for more detail

[1]: https://developers.google.com/android/guides/setup

File: app/src/release/res/values/google_maps_api.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+https://console.developers.google.com/flows/enableapi?apiid=maps_android_backend&keyType=CLIENT_SIDE_ANDROID&r=26:94:57:FE:DE:11:18:E3:2B:5E:32:21:9E:D5:DD:71:CA:16:CC:6E%3Borg.fossasia.openevent.fragments
+
+    You can also add your credentials to an existing key, using this line:
+    26:94:57:FE:DE:11:18:E3:2B:5E:32:21:9E:D5:DD:71:CA:16:CC:6E;org.fossasia.openevent.fragments
+
+    Once you have your key (it starts with "AIza"), replace the "google_maps_key"
+    string in this file.
+    -->
+    <string name="google_maps_key" translatable="false" templateMergeStrategy="preserve">
+        YOUR_KEY_HERE
```
Comment (creativepsyco) [inline app/src/release/res/values/google_maps_api.xml]: ??

**Comment by creativepsyco** [comment]
@mananwason 
Also you would need to add the debug key hash on Google Project to get the API key. This would be a good time to finalise a single debug key to used for debug purposes, can you add me to the Google API Project I will check in a debug key with the finger print.

File: app/build.gradle (hunk 1 above)
Comment (mananwason) [inline app/build.gradle]: I'll change this

**Comment by mananwason** [comment]
@creativepsyco I did added you to the project. I think I have the added the key. Could you please check.

File: app/src/release/res/values/google_maps_api.xml (hunk 2 above)
Comment (mananwason) [inline app/src/release/res/values/google_maps_api.xml]: This is the release key file. Should I add the same key here as well?

**Comment by creativepsyco** [comment]
Ok I got the invite, you fix up the issues here, I shall check in a debug key which we can share.

**Comment by mananwason** [comment]
Cool :+1:

File: app/src/release/res/values/google_maps_api.xml (hunk 2 above)
Comment (creativepsyco) [inline app/src/release/res/values/google_maps_api.xml]: Nah it's okay I will fix that, you commit a working version of Maps for
Debug. Once it's okay lets merge it.

On Thu, Jun 11, 2015 at 6:03 PM, Manan Wason <notifications@github.com>
wrote:

> In app/src/release/res/values/google_maps_api.xml
> <https://github.com/fossasia/open-event-android/pull/27#discussion_r32206250>
> :
>
> > +<resources>
> > +    
> > +    <string name="google_maps_key" translatable="false" templateMergeStrategy="preserve">
> > +        YOUR_KEY_HERE
>
> This is the release key file. Should I add the same key here as well?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fossasia/open-event-android/pull/27/files#r32206250>.
>



-- 
Mohit Singh KANWAL (Mr)

[image: Facebook] <http://facebook.com/mohitkanwal>  [image: Twitter]
<http://twitter.com/mohitkanwal>

**Comment by championswimmer** [comment]
Weren't we thinking of using OSM instead of Google maps ? 

You can check these https://github.com/fossasia/ots15-companion/commit/e85e7708134ba9297be77377d6e70b68b97090ab

https://github.com/fossasia/ots15-companion/commit/28d68b0b0e5dd2413f0d33be5013500625ad8ab4

OSM works just as good as Google Maps. 
Also note, 
We cannot keep the project GPL v3 if we are using GMaps

File: app/build.gradle [hunk 3]
```diff
@@ -21,14 +21,15 @@ android {
...
     compile 'com.android.support:cardview-v7:22.2.0'
-    compile 'com.android.support:recyclerview-v7:+'
+    compile 'com.android.support:recyclerview-v7:22.2.0'
     compile 'com.google.code.gson:gson:2.3.1'
     compile 'com.squareup.okhttp:okhttp:2.0.0'
     compile 'com.squareup.retrofit:retrofit:1.9.0'
     compile 'com.android.support:design:22.2.0'
     compile 'com.squareup.picasso:picasso:2.5.2'
-
+    compile 'com.google.android.gms:play-services-maps:7.5.0'
```
Comment (creativepsyco) [inline app/build.gradle]: Need the base one as well.

**Comment by creativepsyco** [comment]
Why can't we keep GPL v3 if we use GMaps?

On Thu, Jun 11, 2015 at 7:26 PM, Arnav Gupta <notifications@github.com>
wrote:

> Weren't we thinking of using OSM instead of Google maps ?
>
> You can check these fossasia/ots15-companion@e85e770
> <https://github.com/fossasia/ots15-companion/commit/e85e7708134ba9297be77377d6e70b68b97090ab>
>
> fossasia/ots15-companion@28d68b0
> <https://github.com/fossasia/ots15-companion/commit/28d68b0b0e5dd2413f0d33be5013500625ad8ab4>
>
> OSM works just as good as Google Maps.
> Also note,
> We cannot keep the project GPL v3 if we are using GMaps
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fossasia/open-event-android/pull/27#issuecomment-111089890>
> .
>



-- 
Mohit Singh KANWAL (Mr)

[image: Facebook] <http://facebook.com/mohitkanwal>  [image: Twitter]
<http://twitter.com/mohitkanwal>

**Comment by mananwason** [comment]
@creativepsyco isn't open source.

**Comment by mananwason** [comment]
So what should I do @championswimmer @creativepsyco ?

**Comment by creativepsyco** [comment]
ok in that case we will need 2 flavors of the APK one with google maps and one without, I don't know if F droid will have issues with it coz I never submitted an APK there. But the Google Maps dependency will only apply to 1 flavor. I will add a wrapper class with Google Maps and OSM included and the gradle file splitting the flavors. Let's

**Comment by mananwason** [comment]
:+1:

**Comment by championswimmer** [comment]
F-Droid does not accept apps with G Maps.

It for submission to G Maps that I wrote the OSM implementation and created
the fdroid branch in ots-companion repo.

**Comment by creativepsyco** [comment]
Thats okay we will use gradle flavors instead of maintaining separate
branches.

On Thu, Jun 11, 2015 at 9:01 PM, Arnav Gupta <notifications@github.com>
wrote:

> F-Droid does not accept apps with G Maps.
>
> It for submission to G Maps that I wrote the OSM implementation and created
> the fdroid branch in ots-companion repo.
>
> On 11 June 2015 at 18:17, Manan Wason <notifications@github.com> wrote:
>
> > [image: :+1:]
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/fossasia/open-event-android/pull/27#issuecomment-111121412
> >
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fossasia/open-event-android/pull/27#issuecomment-111124365>
> .
>



-- 
Mohit Singh KANWAL (Mr)

[image: Facebook] <http://facebook.com/mohitkanwal>  [image: Twitter]
<http://twitter.com/mohitkanwal>

**Comment by championswimmer** [comment]
The aar of the dependencies (i.e. google maps here) will exist inside the
apk, which will still be a problem then isn't it ?

**Comment by creativepsyco** [comment]
Nope it does not. Different flavors have different dependencies, the final
product will not contain any Google dependencies.

On Thu, Jun 11, 2015 at 9:23 PM, Arnav Gupta <notifications@github.com>
wrote:

> The aar of the dependencies (i.e. google maps here) will exist inside the
> apk, which will still be a problem then isn't it ?
>
>
> On 11 June 2015 at 18:33, Mohit Kanwal <notifications@github.com> wrote:
>
> > Thats okay we will use gradle flavors instead of maintaining separate
> > branches.
> >
> > On Thu, Jun 11, 2015 at 9:01 PM, Arnav Gupta <notifications@github.com>
> > wrote:
> >
> > > F-Droid does not accept apps with G Maps.
> > >
> > > It for submission to G Maps that I wrote the OSM implementation and
> > created
> > > the fdroid branch in ots-companion repo.
> > >
> > > On 11 June 2015 at 18:17, Manan Wason <notifications@github.com>
> wrote:
> > >
> > > > [image: :+1:]
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > >
> >
> https://github.com/fossasia/open-event-android/pull/27#issuecomment-111121412
> > > >
> > >
> > > > .
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
> https://github.com/fossasia/open-event-android/pull/27#issuecomment-111124365
> > >
> > > .
> > >
> >
> >
> >
> > --
> > Mohit Singh KANWAL (Mr)
> >
> > [image: Facebook] <http://facebook.com/mohitkanwal> [image: Twitter]
> > <http://twitter.com/mohitkanwal>
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/fossasia/open-event-android/pull/27#issuecomment-111124854
> >
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fossasia/open-event-android/pull/27#issuecomment-111134298>
> .
>



-- 
Mohit Singh KANWAL (Mr)

[image: Facebook] <http://facebook.com/mohitkanwal>  [image: Twitter]
<http://twitter.com/mohitkanwal>

**Pull request closed without merging**